README.md

```diff
@@ -1,32 +1,193 @@
+    } // <- end of third scope
+
+    else { // <- fourth scope
+      int elseBlockVariable = 4;
+      System.out.printf("else block scope: %d %d %s %d %n" , elseBlockVariable, methodLevelVariable, Arrays.toString(args), classLevelVariable);
+
+    } // <- end of fourth scope
+
+    System.out.printf("main block scope:  %d %s %d %n" , methodLevelVariable, Arrays.toString(args), classLevelVariable);
+
+    anotherMethod(5);
+
+  } // <- end of second scope
+
+  public static void anotherMethod(int parameterVariable) {  // <- fifth scope
+    System.out.printf("anotherMethod block scope: %d %d %n" , classLevelVariable, parameterVariable);
+  }  // <- end of fifth scope
+
+} // <- end of first scope
+```
+
+As we discussed before, curly brace "blocks" can be contained inside each other
+like Russian dolls, which is the case here:
+
+- The first scope is the top level scope, which means nothing can be defined
+  outside it for our program.  
+- The second scope is inside the first scope, which means the second scope has
+  access to all the variables defined in the first scope
+- The third scope is inside the second scope, which means the third scope has
+  access to all the variables defined in the second scope, including the ones
+  defined in the first scope
+- The fourth scope is also inside the second scope, which means the fourth scope
+  has access to all the variables defined in the second scope, including the
+  ones defined in the first scope
+- The fourth scope is a peer of the third scope, which means they cannot see
+  each other's variables. A scope can only see its variables and the variables
+  of all the enclosing scopes.
+- The fifth scope is inside the first scope and a peer to the second scope,
+  which means the fifth scope has access to all the variables defined in the first
+  scope but cannot see the variables in the second scope.
+
+In our example, the variables `ifBlockVariable` and `elseBlockVariable`
+are each declared within their respective part of the `if-else` statement, so they
+are not accessible outside either the `if` or the `else`.
+
+If you run the program enough times it should eventually execute both branches
+of the `if-else` statement, printing the variables in scope for the current block:
+
+```java
+if block scope: 3 2 [] 1 
+main block scope:  2 [] 1 
+anotherMethod block scope: 1 5 
+```
+
+
+```java
+else block scope: 4 2 [] 1 
+main block scope:  2 [] 1 
+anotherMethod block scope: 1 5 
+```
+
+## Shadowing
+
+- A *simple* name is a single identifier, such as `age` or `ssn`.
+- A *qualified* name consists of a name, a `.`, and an identifier, such as `this.age` or `employee1.ssn`.
+
+Shadowing happens a variable declared in one scope (the nested scope)
+has the same name as a variable declared in the enclosing scope.
+In this case, the declaration in the nested scope
```

Comment:
Maybe move this sentence (and the one after it) after showing the code example?

Reply:
Moved and further explained.